shell.nix

```diff
@@ -0,0 +1,8 @@
+# Configuration file for Nix(OS) to provide a temporary development enviroment
```

Comment:
Should we put this file in the repo?

Reply:
Yes it's for development on NixOS with explanation of what it is doing

(will provide a temporary shell environment with needed dependencies)